### PR TITLE
Blog post wjax part 2

### PR DESCRIPTION
**Pull request by BirgitBader**
@skoenig-epages please review.

### CODE REVIEW

File: _posts/2015/2015-12-08-wjax2015-microservices.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+With this follow-up article of the W-JAX retrospect, I'd like to concentrate on the microservices part of the conference.
+
+# Living with microservices
+
+The concepts to develop a software change regularly.
+Some of these changes do not make sense.
+They do not helps us in the development process and are only a refreshed concept of a dusted software idea: Give them a buzzword, talk about it, and everyone will use it.
+
+However a few of these new software design concepts are meaningful and solve current software development problems.
```
Comment (skoenig-epages) [inline _posts/2015/2015-12-08-wjax2015-microservices.md]: Missing comma after However

File: _posts/2015/2015-12-08-wjax2015-microservices.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+One of them is the idea of microservices.
+The participants of the W-JAX 2015 conference talked about this idea in many sessions.
```
Comment (skoenig-epages) [inline _posts/2015/2015-12-08-wjax2015-microservices.md]: participants = speakers

File: _posts/2015/2015-12-08-wjax2015-microservices.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+If two components have to communicate very often both should be in the same microservice.
+- Should not influence another microservice.
+- Every microservice can stand alone without knowing other components of the software.
+- Complete microservice should be handled by only one team.
+If the sliced software component cannot be serviced by one team alone, slice it again.
+
+## Challenges and hints
+
+Microservices will put the weight of complexity from algorithm complexity to communication complexity.
+The dependencies of each software part is now moved to the way of communication.
```
Comment (skoenig-epages) [inline _posts/2015/2015-12-08-wjax2015-microservices.md]: is moved = are moved

File: _posts/2015/2015-12-08-wjax2015-microservices.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+A second big challenge can be the [Two-Phase Commit](http://www.enterpriseintegrationpatterns.com/ramblings/18_starbucks.html) and data consistency.
+Let's imagine we want to transfer money from one to another bank account.
```
Comment (skoenig-epages) [inline _posts/2015/2015-12-08-wjax2015-microservices.md]: from one bank account to another

File: _posts/2015/2015-12-08-wjax2015-microservices.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+The monolith would have a controller layer which sends a request to the model layout.
+This layer will transfer the money from one to another account instantly.
+In a microservice world every bank account can be in another microservice.
+Microservice 1 will send the money to microservice 2 and decrease the money in its own database.
+In this moment the money is not existent: neither in microservice 1 nor in microservice 2.
```
Comment (skoenig-epages) [inline _posts/2015/2015-12-08-wjax2015-microservices.md]: How is this problem solved? Is this a problem at all? --> Eventual consistency

File: _posts/2015/2015-12-08-wjax2015-microservices.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+A very important concept is the **Design for Failure**.
+With different services the input values of every service should be mistrusted.
```
Comment (skoenig-epages) [inline _posts/2015/2015-12-08-wjax2015-microservices.md]: What does this imply? What to do with input values?

**Comment by BirgitBader** [comment]
@dpauli please check the feedback with @skoenig-epages

File: _posts/2015/2015-12-08-wjax2015-microservices.md (hunk 5 above)
Comment (dpauli) [inline _posts/2015/2015-12-08-wjax2015-microservices.md]: I don't want to talk about solutions of microservice problems, but about the content of the conference ;)

File: _posts/2015/2015-12-08-wjax2015-microservices.md (hunk 6 above)
Comment (dpauli) [inline _posts/2015/2015-12-08-wjax2015-microservices.md]: I don't want to talk about solutions of microservice problems, but about the content of the conference ;)

File: _posts/2015/2015-12-08-wjax2015-microservices.md (hunk 5 above)
Comment (skoenig-epages) [inline _posts/2015/2015-12-08-wjax2015-microservices.md]: The talk "Jave EE meets Microservices" covered these topics and explained, what eventual consistency is and that you cannot guarantee immediate consistency in distributed environments. See also http://www.enterpriseintegrationpatterns.com/ramblings/18_starbucks.htm

**Comment by BirgitBader** [comment]
Closed in favour of #310

**Pull request closed without merging**